incapsula/resource_siem_connnection_s3.go

```diff
@@ -0,0 +1,195 @@
+package incapsula
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceSiemConnectionS3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSiemConnectionS3Create,
+		Read:   resourceSiemConnectionS3Read,
+		Update: resourceSiemConnectionS3Update,
+		Delete: resourceSiemConnectionS3Delete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.SetId(d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Description: "Client account id.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"connection_name": {
+				Description: "Name of the connection.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"storage_type": {
+				Description: "Type of the storage.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if val.(string) != "CUSTOMER_S3" {
+						errs = append(errs, fmt.Errorf("expected %s value for %s, got %s", "CUSTOMER_S3", "storage_type", val.(string)))
+					}
+					return
+				},
+				DefaultFunc: func() (interface{}, error) {
+					return "CUSTOMER_S3", nil
+				},
+			},
+			"access_key": {
```

Comment:
what about the RequiredWith: validation we talk about?

Reply:
Adding RequiredWith validation on field level will improve the validation checks, so I will at them. Concerning storate_type here, it is stated that this field is required and could have only CUSTOMER_S3 value, with initial (default) value if not provided, set to CUSTOMER_S3.

Reply:
didn't we agree to totally remove the storage_type attribute?